collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormDownloader.java

```diff
@@ -58,14 +58,16 @@ public ServerFormDownloader(FormSource formSource, FormsRepository formsReposito
 
     @Override
     public void downloadForm(ServerFormDetails form, @Nullable ProgressReporter progressReporter, @Nullable Supplier<Boolean> isCancelled) throws FormDownloadException, InterruptedException {
-        Form formOnDevice = formsRepository.getOneByFormIdAndVersion(form.getFormId(), form.getFormVersion());
+        Form formOnDevice = formsRepository.getLatestByFormIdAndVersion(form.getFormId(), form.getFormVersion());
```

Comment:
I think maybe we need a test for this change in `ServerFormDownloaderTest`? This is needed before if we download a new form with the same version and id (but a different hash) we end up with two forms in the repository right?

Reply:
Fair enough. In writing the test, I realized we have access to the hash there so there's no need to use a heuristic. I verified that everywhere else we use `getLatestByFormIdAndVersion` there's nothing better we can do because we're dealing with an instance.

I think we should really start tracking the form hash an instance was instantiated from.